tool/tsh/kube.go

```diff
@@ -582,18 +586,30 @@ func newKubeCredentialsCommand(parent *kingpin.CmdClause) *kubeCredentialsComman
 }
 
 func (c *kubeCredentialsCommand) run(cf *CLIConf) error {
+	profile, err := cf.GetProfile()
+	if err != nil {
+		// Cannot find the profile, continue to c.issueCert for a login.
+		return trace.Wrap(c.issueCert(cf))
+	}
+
+	if err := c.checkLocalProxyRequirement(profile.TLSRoutingConnUpgradeRequired); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Out of the scope of this PR, just a feature idea.
I wonder if we can add support for raw `kubectl` flow without bothering a user to start a local proxy manually. 

For instance, if the proxy is required we can start it automatically under the hood of execution of `tsh kube credentials` with some TTL like 30s, and the next call of `kubectl` will check if a local proxy is running by doing a "locking" health check that will prolong the expiration time by +30s and reuse local proxy process or just create a new kube local proxy in background. 

For sure this approach will somehow impact performance but I think that it is worth considering if we want to sacrifice a few ms for a better UX experience. 

@tigrato @greedy52 What do you think about this approach ?

Reply:
`tsh kube credential` is used to retrieve user certs. I believe `tsh kube login` has to generate different cluster configs in the default kubeconfig in order for kubectl to talk to the local proxy. And it may have to pre-select a port. And the local proxy has to be detached and coordinate parallel calls? so might need to implement some IPC like a mini webserver.